### PR TITLE
Fix issue 1122

### PR DESCRIPTION
**Pull request by hanabi1224**
Fix  https://github.com/Azure/azure-iot-sdk-csharp/issues/1122
It fails when default JsonSerializerSettings.DateParseHandling is set to DateParseHandling.DateTimeOffset

### CODE REVIEW

**Comment by hanabi1224** [comment]
I believe a better fix would be changing all DateTime usage to DateTimeOffset given that timezone info is part of timezone string in actual twin

**Comment by abhipsaMisra** [comment]
Thank you for submitting this PR @hanabi1224, I'll take a look at it. 
In the meanwhile, could you rebase your branch off of `master`, so that you bring in the latest changes!

**Comment by hanabi1224** [comment]
@abhipsaMisra Thanks! It’s now up to date

**Comment by hanabi1224** [comment]
@abhipsaMisra Kindly ping, any feedback?

**Comment by abhipsaMisra** [comment]
@hanabi1224 Thanks for your patience with this, reviewing it now.

File: common/src/service/HttpClientHelper.cs [hunk 1]
```diff
@@ -24,6 +25,7 @@ namespace Microsoft.Azure.Devices
 
     sealed class HttpClientHelper : IHttpClientHelper
     {
+        static readonly JsonSerializerSettings s_jsonSerializerSettings = new JsonSerializerSettings { DateParseHandling = DateParseHandling.DateTime, };
```
Comment (abhipsaMisra) [inline common/src/service/HttpClientHelper.cs]: The DateParseHandling property has a default value of DateTime. Could you please explain the advantage of setting it explicitly as above?

Comment (hanabi1224) [reply common/src/service/HttpClientHelper.cs]: @abhipsaMisra The problem is that, user can override default settings,  as described in https://github.com/Azure/azure-iot-sdk-csharp/issues/1122

Comment (abhipsaMisra) [reply common/src/service/HttpClientHelper.cs]: Are you setting the json serialization settings in your application? Can you share a code snippet that reproduces this issue.

Comment (hanabi1224) [reply common/src/service/HttpClientHelper.cs]:
```csharp
JsonConvert.DefaultSettings = () =>new JsonSerializerSettings
            {
                DateParseHandling = DateParseHandling.DateTimeOffset,
            };
```

**Comment by abhipsaMisra** [comment]
This issue requires some further investigation on if we should support DateTimeOffset directly in the twin structure returned, and throughout (based on the service response). As you've pointed out above, that would be a better fix rather than forcing all applications to use DateTime mandatorily.
I'll create a separate PR addressing this bigger issue, and close this PR for now.

**Comment by abhipsaMisra** [comment]
Closing this PR. This issue will be addressed in #1175

**Comment by abhipsaMisra** [comment]
Thank you for

**Comment by abhipsaMisra** [comment]
Thank you for highlighting this issue and submitting the PR @hanabi1224 .

This issue requires some further investigation on if we should support DateTimeOffset directly in the twin structure returned, and throughout (based on the service response). As you've pointed out above, that would be a better fix rather than forcing all applications to use DateTime mandatorily.

I've created a separate PR #1175 to fix this; you can use this to track this issue.

**Pull request closed without merging**